service/service.go

```diff
@@ -177,38 +177,38 @@ func (s *Service) loop(ctx context.Context) {
 	var executingRounds errgroup.Group
 	defer executingRounds.Wait()
 
-	timer := time.NewTimer(0)
-	<-timer.C
-	defer timer.Stop()
 	for {
-		start := s.genesis.
-			Add(s.cfg.EpochDuration * time.Duration(s.openRound.Epoch())).
-			Add(s.cfg.PhaseShift)
-		if d := time.Until(start); d > 0 {
-			log.Info("Round %v waiting for execution to start for %v", s.openRoundID(), d)
-			timer.Reset(d)
-			select {
-			case <-timer.C:
-			case <-ctx.Done():
-				log.Info("service shutting down")
-				s.openRoundMutex.Lock()
-				s.openRound = nil
-				s.openRoundMutex.Unlock()
-				return
-			}
+		s.openRoundMutex.RLock()
+		epoch := s.openRound.Epoch()
+		s.openRoundMutex.RUnlock()
+
+		start := s.genesis.Add(s.cfg.EpochDuration * time.Duration(epoch)).Add(s.cfg.PhaseShift)
+		waitTime := time.Until(start)
+		if waitTime > 0 {
+			log.Info("Round %v waiting for execution to start for %v", s.openRoundID(), waitTime)
+		}
+		select {
+		case <-time.After(waitTime):
```

Comment:
This will trigger at `start` but probably some time slightly earlier.

Reply:
It doesn't really matter if it triggers 1ms before. What's the right way to write this code in you opinion?